### ISSUE TITLE
[BUG] Prefix from tailwind.config.js not working with named groups

### ISSUE THREAD

**Issue by bezbac** [francoismassart/eslint-plugin-tailwindcss#243]
**Describe the bug**
A colleague of mine discovered this issue. We use the ESlint plugin at work and have a shared component library using a tailwind with a prefix to avoid collisions with other tailwind projects. While developing a new feature, we wanted to use the recently introduced syntax for named groups, but ESlint kept nagging about a non-existent class.
I looked into the code for `no-custom-classname`, and the regex responsible doesn't include the prefix from the tailwind config. I believe this is the issue.

**To Reproduce**
```
// HTML
<div class="tw-group/edit">
  <div class="group-hover/edit:tw-text-red-500">Some content</div>
</div>


// tailwind.config.js
/** @type {import('tailwindcss').Config} */
module.exports = {
  prefix: "tw-"
}
``` 

The plugin will report a false positive unknown class name `tw-group/edit` and `group-hover/edit:tw-text-red-500`

**Expected behavior**
Support the specified prefix specified inside the `tailwind.config.js` together with the named groups and peer syntax.

**Comment by francoismassart** [comment]
Available for testing and feeback via `npm i eslint-plugin-tailwindcss@3.12.1-beta.0`
